lib/ngannotate/processor.rb

```diff
@@ -29,10 +29,14 @@ def force
     # Skip processing in environments where it does not make sense.
     # Override by NG_FORCE=true env variable
     #
-    def skip
+    def skip_by_config
       !force && !::Rails.configuration.ng_annotate.process
     end
 
+    def skip
+      skip_by_config || !@file.start_with?(::Rails.root.to_s)
```

Comment:
> I think I figured out possible fix, instead of checking for rails root dir, can check for "/vendor/" subpath existence in filename, and it should be able to cover various cases correctly.

I'm not sure it's the correct way
If you require js from gems (for example rails-assets) it could be inside any directory

> And naturally assets owned by engine are not under rails root dir, but within "gems" directory.

oh, i see
you're right
In this case - lets move this to configuration ? for example Rails.root.to_s would be by default and you can change it